runbotics/runbotics-orchestrator-ui/src/src-app/Actions/folder.actions.ts

```diff
@@ -161,6 +161,82 @@ const getFolderActions = (): Record<string, IBpmnAction> => ({
             },
         },
     },
+    'folder.rename': {
+        id: FolderAction.RENAME,
+        label: translate('Process.Details.Modeler.Actions.Folder.Rename.Label'),
+        script: FolderAction.RENAME,
+        runner: Runner.DESKTOP_SCRIPT,
+        output: {
+            assignVariables: true,
+            outputMethods: {
+                variableName: '${content.output[0]}',
+            },
+        },
+        form: {
+            schema: {
+                type: 'object',
+                properties: {
+                    input: {
+                        title: translate('Process.Details.Modeler.Actions.Folder.Rename.Label'),
+                        type: 'object',
+                        properties: {
+                            path: {
+                                title: translate('Process.Details.Modeler.Actions.Folder.Path'),
+                                type: 'string',
+                            },
+                            newName: {
+                                title: translate('Process.Details.Modeler.Actions.Folder.Rename.NewName'),
+                                type: 'string',
+                            },
```

Comment:
you can specify pattern (just like it's done with pattern: ActionRegex.VARIABLE_NAME few lines below) to inform the user about using illegall characters for directory name or path - better UX